### PR TITLE
Backward iteration for termdict range

### PR DESCRIPTION
**Pull request by halvorboe**
Depending on https://github.com/tantivy-search/fst/pull/12.

Solves https://github.com/tantivy-search/tantivy/issues/689.

### CODE REVIEW

**Comment by fulmicoton** [comment]
Make sure CI passes. Here the issue is that you need to format your code apparently
 `cargo fmt --all`.

**Comment by halvorboe** [comment]
Fixed it now. Still need to merge the PR in fst and update the dependency.

**Comment by fulmicoton** [comment]
I don't understand the change from backward to backwards.
From what I understand backward is the US en and backwards is british en.
https://www.quickanddirtytips.com/education/grammar/backward-versus-backwards

Also, assuming the change of name is a good idea, this breaks API and requires to inc the version to `0.3.0`, not `0.2.1`.

**Comment by fulmicoton** [comment]
See comment in the tantivy-fst PR conversaiton tab.

File: src/termdict/mod.rs [hunk 1]
```diff
@@ -352,41 +352,91 @@ mod tests {
         let source = ReadOnlySource::from(buffer);
         let term_dictionary: TermDictionary = TermDictionary::from_source(&source);
 
-        let value_list = |mut streamer: TermStreamer<'_>| {
+        let value_list = |mut streamer: TermStreamer<'_>, backwards: bool| {
```
Comment (fulmicoton) [inline src/termdict/mod.rs]: I'd remove the `backwards: bool`  arg here and compute both backward and forward.
Reverse the backward one, and assert_eq! that it is equal to forward, and then return the result.

Note that it is not considered a good practies in the industry to make non atomic unit test. I don't think it is necessary to observe this practise in a project like tantivy.

**Comment by halvorboe** [comment]
My bad. Googled it after making the change. Turns out backward is "as correct" as backwards - just looked really wrong to me. Reverted the change.

File: Cargo.toml [hunk 2]
```diff
@@ -18,7 +18,7 @@ byteorder = "1.0"
 crc32fast = "1.2.0"
 once_cell = "1.0"
 regex ={version = "1.3.0", default-features = false, features = ["std"]}
-tantivy-fst = "0.2"
+# tantivy-fst = "0.2"
```
Comment (fulmicoton) [inline Cargo.toml]: I merged and published tantivy-fst = 0.2.1.

Can you update this and remove `[dependencies.tantivy-fst]`

**Comment by fulmicoton** [comment]
LGTM

Can you:
- update `Cargo.toml` to use tantivy-fst = 0.2.1 directly
- add your change in the CHANGELOG, (and don't forget to mention your name!)

File: Cargo.toml (hunk 2 above)
Comment (fulmicoton) [inline Cargo.toml]: I merged and published tantivy-fst = 0.2.1.

Can you update this and remove [dependencies.tantivy-fst]

**Review by fulmicoton — CHANGES_REQUESTED** [review]
LGTM

Can you:

update Cargo.toml to use tantivy-fst = 0.2.1 directly
add your change in the CHANGELOG, (and don't forget to mention your name!)

File: Cargo.toml (hunk 2 above)
Comment (halvorboe) [reply Cargo.toml]: Done :D

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+Tantivy 0.12.1
```
Comment (fulmicoton) [inline CHANGELOG.md]: It will part of 0.12.0.

File: Cargo.toml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "tantivy"
-version = "0.11.3"
+version = "0.12.1"
```
Comment (fulmicoton) [inline Cargo.toml]: 0.12.0

**Review by fulmicoton — CHANGES_REQUESTED** [review]

**Comment by fulmicoton** [comment]
Awesome thank you @halvorboe!

**Pull request merged**